File: BoardGameGeek/src/com/boardgamegeek/provider/PublishersIdProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Publishers;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class PublishersIdProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int publisherId = Publishers.getPublisherId(uri);
		return new SelectionBuilder().table(Tables.PUBLISHERS).whereEquals(Publishers.PUBLISHER_ID, publisherId);
	}

	@Override
	protected String getPath() {
		return "publishers/#";
	}

	@Override
	protected String getType(Uri uri) {
		return Publishers.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/PublishersIdProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.Companion.PATH_PUBLISHERS
import com.boardgamegeek.provider.BggContract.Publishers
import com.boardgamegeek.provider.BggDatabase.Tables

class PublishersIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Publishers.CONTENT_ITEM_TYPE

    override val path = "$PATH_PUBLISHERS/#"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val publisherId = Publishers.getPublisherId(uri)
        return SelectionBuilder()
            .table(Tables.PUBLISHERS)
            .whereEquals(Publishers.Columns.PUBLISHER_ID, publisherId)
    }
}
